common/src/main/java/org/dash/wallet/common/ui/receive/ReceiveInfoView.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.ui.receive
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Context
+import android.graphics.drawable.BitmapDrawable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import org.bitcoinj.core.Address
+import org.bitcoinj.core.Coin
+import org.bitcoinj.uri.BitcoinURI
+import org.bitcoinj.uri.BitcoinURIParseException
+import org.dash.wallet.common.R
+import org.dash.wallet.common.databinding.ReceiveInfoViewBinding
+import org.dash.wallet.common.util.Qr
+import org.dash.wallet.common.util.shareText
+import org.slf4j.LoggerFactory
+
+
+class ReceiveInfoView(context: Context, attrs: AttributeSet?) : ConstraintLayout(context, attrs) {
```

Comment:
This class seems sufficiently generic and reusable to be moved to the `common`; I have modified it to remove some hard `wallet` dependencies.